from .function import Function

from .error import Error

### FILE IS AUTOGENERATED. DO NO EDIT IT BY HAND ###

class TestReturnError(Function[Error]):
    """
    Returns the specified error and ensures that the Error object is used; for testing only. Can be called synchronously

    :param error: The error to be returned
    :return: :class:`Error`
    """
    __slots__ = ("error", "_extra", "_client_id", "_type")

    def __init__(self, error: Error | None = None):
        self.error = error
        self._type = "testReturnError"